source/windows/bcrypt_aes.c

```diff
@@ -0,0 +1,1110 @@
+            cipher->auth_info_ptr->cbAuthData = (ULONG)cipher->cipher.aad.len;
+        }
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+/* Free up as few resources as possible so we can quickly reuse the cipher. */
+static void s_clear_reusable_components(struct aws_symmetric_cipher *cipher) {
+    struct aes_bcrypt_cipher *cipher_impl = cipher->impl;
+    bool working_iv_optimized = cipher->iv.buffer == cipher_impl->working_iv.buffer;
+
+    if (!working_iv_optimized) {
+        aws_byte_buf_secure_zero(&cipher_impl->working_iv);
+    }
+
+    /* These can't always be reused in the next operation, so go ahead and destroy it
+       and create another. */
+    if (cipher_impl->key_handle) {
+        BCryptDestroyKey(cipher_impl->key_handle);
+        cipher_impl->key_handle = NULL;
+    }
+
+    if (cipher_impl->auth_info_ptr) {
+        aws_mem_release(cipher->allocator, cipher_impl->auth_info_ptr);
+        cipher_impl->auth_info_ptr = NULL;
+    }
+
+    aws_byte_buf_secure_zero(&cipher_impl->overflow);
+    aws_byte_buf_secure_zero(&cipher_impl->working_mac_buffer);
+    /* windows handles this, just go ahead and tell the API it's got a length. */
+    cipher_impl->working_mac_buffer.len = AWS_AES_256_CIPHER_BLOCK_SIZE;
+}
+
+static int s_reset_cbc_cipher(struct aws_symmetric_cipher *cipher) {
+    struct aes_bcrypt_cipher *cipher_impl = cipher->impl;
+
+    s_clear_reusable_components(cipher);
+    return s_initialize_cipher_materials(
+        cipher_impl, NULL, NULL, NULL, NULL, AWS_AES_256_CIPHER_BLOCK_SIZE, false, false);
+}
+
+static int s_reset_ctr_cipher(struct aws_symmetric_cipher *cipher) {
+    struct aes_bcrypt_cipher *cipher_impl = cipher->impl;
+
+    s_clear_reusable_components(cipher);
+    struct aws_byte_cursor iv_cur = aws_byte_cursor_from_buf(&cipher->iv);
+    /* reset the working iv back to the original IV. We do this because
+       we're manually maintaining the counter. */
+    aws_byte_buf_append_dynamic(&cipher_impl->working_iv, &iv_cur);
+    return s_initialize_cipher_materials(
+        cipher_impl, NULL, NULL, NULL, NULL, AWS_AES_256_CIPHER_BLOCK_SIZE, true, false);
+}
+
+static int s_reset_gcm_cipher(struct aws_symmetric_cipher *cipher) {
+    struct aes_bcrypt_cipher *cipher_impl = cipher->impl;
+
+    s_clear_reusable_components(cipher);
+    return s_initialize_cipher_materials(
+        cipher_impl, NULL, NULL, NULL, NULL, AWS_AES_256_CIPHER_BLOCK_SIZE - 4, false, true);
+}
+
+static int s_aes_default_encrypt(
+    struct aws_symmetric_cipher *cipher,
+    const struct aws_byte_cursor *to_encrypt,
+    struct aws_byte_buf *out) {
+    struct aes_bcrypt_cipher *cipher_impl = cipher->impl;
+
+    if (to_encrypt->len == 0) {
+        return AWS_OP_SUCCESS;
+    }
+
+    size_t predicted_write_length =
+        cipher_impl->cipher_flags & BCRYPT_BLOCK_PADDING
+            ? to_encrypt->len + (AWS_AES_256_CIPHER_BLOCK_SIZE - to_encrypt->len % AWS_AES_256_CIPHER_BLOCK_SIZE)
```

Comment:
utterly trivial: extra parens to make order of operations clear:
```suggestion
            ? to_encrypt->len + (AWS_AES_256_CIPHER_BLOCK_SIZE - (to_encrypt->len % AWS_AES_256_CIPHER_BLOCK_SIZE))
```
right?